confmap/confmap.go

```diff
@@ -306,6 +307,23 @@ func isStringyStructure(t reflect.Type) bool {
 	return false
 }
 
+// safeWrapDecodeHookFunc wraps a DecodeHookFuncValue to ensure that it is safe to call all
+// methods in fromVal.
```

Comment:
```suggestion
// safeWrapDecodeHookFunc wraps a DecodeHookFuncValue to ensure fromVal is a valid `reflect.Value`
// object and therefore it is safe to call `reflect.Value` methods on fromVal.
```
Just a slight clarification since you can use `MethodByName` to call actual methods on the values instead of just the methods on `reflect.Value`.